wave_geometry/src/rotation.cpp

```diff
@@ -66,17 +87,37 @@ Rotation &Rotation::setFromEulerXYZ(const Vec3 &input_vector) {
 
 Rotation &Rotation::setFromAngleAxis(const double angle_magnitude,
                                      const Vec3 &rotation_axis) {
+    // Check if the input is finite, throw error otherwise
+    checkArrayFinite(rotation_axis);
+    checkScalarFinite(angle_magnitude);
+
+    // Check if the input rotation axis is normalized,
+    // throw error otherwise.
+
+    checkVectorNormalized(rotation_axis);
```

Comment:
I think having our own `checkVectorNormalized` is fine for now, I mean the ultimate goal is to eventually have our own version of Kindr right?